core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1020,6 +1020,17 @@ class ReplicaManager(val config: KafkaConfig,
             exception = None)
         }
       } catch {
+        case e@ (_: OffsetOutOfRangeExceptionWithOffsetValues) =>
+          LogReadResult(info = FetchDataInfo(LogOffsetMetadata.UnknownOffsetMetadata, MemoryRecords.EMPTY),
+              highWatermark = Log.UnknownOffset,
```

Comment:
If we are providing last stable offset, we want to provide high watermark as well.

Reply:
refactored